### ISSUE TITLE
Confidentiality level 4 for PSK patterns when sent by an anonymous agent

### ISSUE THREAD

**Issue by symphorien** [SymbolicSoft/noiseexplorer#9]
I am under the impression that the query for confidentiality level 4 for PSK patterns of transport messages sent by an anonymous agent is wrong.

For example, the query for message D of INpsk2 is
https://github.com/SymbolicSoft/noiseexplorer/blob/0581a32bad81f12543d30ebb2ae911f15c8535a5/models/INpsk2.noise.active.pv#L453-L454
This is a stronger statement than the query for level 5:
https://github.com/SymbolicSoft/noiseexplorer/blob/0581a32bad81f12543d30ebb2ae911f15c8535a5/models/INpsk2.noise.active.pv#L455-L456
As a result, message D of INpks2 satisfies level 5 but not level 4, and it is this labeled as level 3 instead of 5 on the website:
https://noiseexplorer.com/patterns/INpsk2/index.html
For comparison, the message D of IKpsk2 and IN are both level 5.
I would have expected something along the lines of
```
attacker(msg_d(bob, alice, sid_b)) ==> ((event(LeakS(phase0, alice))) && (event(LeakPsk(phase0, alice, bob)))) || ((event(LeakS(px, alice))) && (event(LeakPsk(py, alice, bob))) && (false)));
```
(inspired from IKpks2 by replacing `leakS(x, bob)` by false:
https://github.com/SymbolicSoft/noiseexplorer/blob/0581a32bad81f12543d30ebb2ae911f15c8535a5/models/IKpsk2.noise.active.pv#L459-L460
)
This is equivalent to level 5, so it should be true.

This phenomenon happens with all transport messages of PSK handshakes sent by an anonymous agent I saw:
https://github.com/SymbolicSoft/noiseexplorer/blob/0581a32bad81f12543d30ebb2ae911f15c8535a5/models/INpsk1.noise.active.pv#L453-L454
https://github.com/SymbolicSoft/noiseexplorer/blob/0581a32bad81f12543d30ebb2ae911f15c8535a5/models/KNpsk2.noise.active.pv#L451-L452
https://github.com/SymbolicSoft/noiseexplorer/blob/0581a32bad81f12543d30ebb2ae911f15c8535a5/models/XNpsk3.noise.active.pv#L475-L476
etc.

Do you also think this is a bug or is there something I missed?

**Comment by kaepora** [comment]
@symphorien Thanks for opening this issue. We've been debating internally how to properly formalize PSK queries for a little while. I will get back to you soon with a more elaborate response: we are currently doing a lot of internal work on #3 and this is taking up a lot of our focus.

**Comment by kaepora** [comment]
@symphorien Thank you for your patience.

Your finding is 100% relevant and was a very important catch. It has been resolved in dee1d3247074806d0ab07b1d1b843d52861ad5ed and we will be shortly updating the Noise Explorer compendium to correct the misleading results.

Thank you very much for this.
